plugins/Marketplace/Plugins.php

```diff
@@ -247,7 +254,8 @@ private function enrichPluginInformation($plugin)
         $plugin['isActivated']  = $this->isPluginActivated($plugin['name']);
         $plugin['isInvalid']    = $this->pluginManager->isPluginThirdPartyAndBogus($plugin['name']);
         $plugin['canBeUpdated'] = $plugin['isInstalled'] && $this->hasPluginUpdate($plugin);
-        $plugin['lastUpdated'] = $this->toShortDate($plugin['lastUpdated']);
+        $plugin['lastUpdated']  = $this->toShortDate($plugin['lastUpdated']);
+        $plugin['canBePurchased'] = !$plugin['isDownloadable'] && $plugin['shop'] && ($plugin['shop']['url'] ?? false);
```

Comment:
I'd personally prefer
```suggestion
        $plugin['canBePurchased'] = !$plugin['isDownloadable'] && !empty($plugin['shop']['url']);
```